### ISSUE TITLE
Incorrect use of logging module causing poor performance

### ISSUE THREAD

**Issue by jamescasbon** [nipy/nipype#1242]
This code is in engine... 

        logger.debug('setting nodelevel(%s) input %s = %s' % (str(self),
                                                              parameter,
                                                              str(val)))

The `str` calls here are not needed.   The logger should call str *if needed*.  By doing this, you pay the cost of string representing inputs even when debug logging is not activated.  For very large objects, this can be significant.

**Comment by oesteban** [comment]
Indeed, we should specify parameters as logger arguments instead of using formatting (http://docs.pylint.org/features.html, W1201)
